test/distributed/checkpoint/test_fsdp_model_state.py

```diff
@@ -43,32 +43,36 @@ def _test_fsdp_model_state(self, process_group) -> None:
                 planner=DefaultSavePlanner(),
             )
 
-        model_2 = FSDP(
-            torch.nn.Linear(8, 8, device="meta"), process_group=process_group
-        )
-
-        with FSDP.summon_full_params(model):
-            with FSDP.summon_full_params(model_2):
-                self.assertNotEqual(model.weight, model_2.weight)
-                self.assertNotEqual(model.bias, model_2.bias)
-
-        # now load the model and ensure the values are the same
-        with FSDP.state_dict_type(model_2, StateDictType.SHARDED_STATE_DICT):
-            state_dict = {
-                "model": model_2.state_dict(),
-            }
-
-            dist_cp.load_state_dict(
-                state_dict=state_dict,
-                storage_reader=dist_cp.FileSystemReader(CHECKPOINT_DIR),
-                planner=DefaultLoadPlanner(),
-            )
-            model_2.load_state_dict(state_dict["model"])
-
-        with FSDP.summon_full_params(model):
-            with FSDP.summon_full_params(model_2):
-                self.assertEqual(model.weight, model_2.weight)
-                self.assertEqual(model.bias, model_2.bias)
+        file_system_reader = dist_cp.FileSystemReader(CHECKPOINT_DIR)
+        metadata = file_system_reader.read_metadata()
+        print(f"{metadata=}")
```

Comment:
After writing the metadata, we should be able to read it back like this. @rohan-varma